macros/blueprint-proc-macro/src/lib.rs

```diff
@@ -70,7 +80,14 @@ pub fn job(args: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
-/// TDOD: Documentation
+/// A procedural macro that annotates a function as a registration hook, mainly used
+/// for the metadata in the `blueprint.json`.
+/// # Example
+/// ```rust,no_run
+/// # use gadget_blueprint_proc_macro::registration_hook;
+/// #[registration_hook(evm = "MyRegistrationHook")]
+/// pub fn my_registration_hook(nft_id: u32);
```

Comment:
```suggestion
/// pub fn my_registration_hook();
```

If we can't describe examples with respect to arguments in our docs, then we shouldn't be putting things like `nft_id` in the docs. It's confusing, I don't know what you intend to show me as a developer what I do with inputting an NFT ID. Where does this ID come from? If we don't have clarity don't add unclear details.